### PR TITLE
some minor changes

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
refactor sparse ndarray constructor. fix warnings for gpu build.

### CODE REVIEW

File: include/mxnet/op_attr_types.h [hunk 1]
```diff
@@ -18,6 +17,11 @@
 #include "./operator.h"
 #include "./ndarray.h"
 
+#define FCOMP_EX_CPU_DEFAULT "FComputeEx<cpu, default>"
```
Comment (reminisce) [inline include/mxnet/op_attr_types.h]: Remove default and non-default as we discussed.

File: src/operator/tensor/elemwise_binary_op.h [hunk 2]
```diff
@@ -35,7 +36,7 @@ void BinaryCompute(const nnvm::NodeAttrs& attrs,
 // TODO(haibin) This is an inefficient temporary implementation
 // Binary Compute between two row-sparse ndarray
 template<typename xpu, typename OP>
-void BinaryComputeExRsRs(const nnvm::NodeAttrs& attrs,
+void BinaryComputeRsRs(const nnvm::NodeAttrs& attrs,
```
Comment (reminisce) [inline src/operator/tensor/elemwise_binary_op.h]: Maybe naming it BinaryComputeRspRsp is better?

File: src/operator/tensor/elemwise_unary_op.h [hunk 3]
```diff
@@ -150,7 +160,7 @@ struct CastStorageParam : public dmlc::Parameter<CastStorageParam> {
 };
 
 template<typename xpu>
-void CastStorageComputeEx(const nnvm::NodeAttrs& attrs,
+void CastStorageComputeRspDsn(const nnvm::NodeAttrs& attrs,
```
Comment (reminisce) [inline src/operator/tensor/elemwise_unary_op.h]: Dsn --> Dns

File: src/operator/tensor/elemwise_binary_op.h (hunk 2 above)
Comment (eric-haibin-lin) [inline src/operator/tensor/elemwise_binary_op.h]: yes

File: src/operator/tensor/elemwise_unary_op.h (hunk 3 above)
Comment (eric-haibin-lin) [inline src/operator/tensor/elemwise_unary_op.h]: good catch

**Comment by eric-haibin-lin** [comment]
@reminisce I also updated the none value for storage type to be kUndefinedStorage during FInferStorage. passed python/unittest and python/train. merging it in now

**Pull request merged**